iati_datastore/iatilib/crawler.py

```diff
@@ -17,94 +17,41 @@
 
-CKAN_WEB_BASE = 'https://iatiregistry.org/dataset/%s'
-CKAN_API = 'https://iatiregistry.org'
-
-registry = ckanapi.RemoteCKAN(CKAN_API, get_only=True)
 
 manager = Blueprint('crawler', __name__)
 manager.cli.short_help = "Crawl IATI registry"
 
 
-class CouldNotFetchPackageList(Exception):
-    pass
-
-
-def fetch_dataset_list(modified_since=None):
+def fetch_dataset_list():
     '''
-    Fetches datasets from CKAN and stores them in the DB. Either passed a datetime to filter on more recently modified
-    datasets, or updates the entire set. Used in update() to update the Flask job queue. Uses CKAN metadata to determine
-    if an activity is active or deleted, and tries to determine if a dataset is actually new or not.
-    :param modified_since:
+    Fetches datasets from iatikit and stores them in the DB. Used in update() to update the Flask job queue. Uses CKAN metadata to determine
+    if an activity is active or deleted.
     :return:
     '''
     existing_datasets = Dataset.query.all()
-    existing_ds_names = set(ds.name for ds in existing_datasets)
-    if modified_since:
-        solr_date_format = modified_since.strftime('%Y-%m-%dT%H:%M:%SZ')
-        try:
-            search_result = registry.action.package_search(
-                    fq='metadata_modified:[{0} TO NOW]'.format(solr_date_format),
-            )
-        except Exception:
-            raise CouldNotFetchPackageList()
-        step = 50
-        count = search_result['count']
-        package_list = []
-        for current_start in range(0, count, step):
-            packages = registry.action.package_search(
-                    fq='metadata_modified:[{0} TO NOW]'.format(solr_date_format),
-                    start=current_start,
-                    rows=step,
-
-            )
-            package_list = package_list + packages['results']
-
-        incoming_ds_names = set()
-        deleted_ds_names = set()
-
-        for dataset in package_list:
-            if dataset['state'] == 'active':
-                incoming_ds_names.add(dataset['name'])
-            elif dataset['state'] == 'deleted':
-                deleted_ds_names.add(dataset['name'])
+    existing_ds_names = set((ds.publisher, ds.name) for ds in existing_datasets)
 
-        # TODO check if the following line is the source of activity mismatch.
-        new_datasets = [Dataset(name=n) for n
-                        in incoming_ds_names - existing_ds_names]
+    package_list = [
+        tuple(x[:-4].rsplit('/', 2)[1:])
+        for x in glob.glob('__iatikitcache__/registry/data/*/*')]
+    incoming_ds_names = set(package_list)
 
-        db.session.add_all(new_datasets)
-        db.session.commit()
-
-        if deleted_ds_names:
-            delete_datasets(deleted_ds_names)
+    new_datasets = [Dataset(name=n, publisher=p) for p, n
+                    in incoming_ds_names - existing_ds_names]
+    all_datasets = existing_datasets + new_datasets
+    last_seen = iatikit.data().last_updated
```

Comment:
Also: I think this is consistent with how things worked before, but is dependent on Registry metadata which we know to be unreliable. Just making a note here and that it would be good to compare hashes?

Reply:
Yeah totally agree. But I think we should come back to this.